src/EasyAbp.Abp.Trees.EntityFrameworkCore/EasyAbp/Abp/Trees/EfCoreTreeRepository.cs

```diff
@@ -139,58 +146,74 @@ public async override Task DeleteAsync(Guid id, bool autoSave = false, Cancellat
 
             await base.DeleteAsync(id, autoSave, cancellationToken);
         }
-        protected virtual async Task<TEntity> GetLastChildOrNullAsync(Guid? parentId)
-        {
-            var children = await this.Where(tree => tree.ParentId == parentId).ToListAsync();
-            return children.OrderBy(c => c.Code).LastOrDefault();
-        }
-        protected virtual async Task<string> GetNextChildCodeAsync(Guid? parentId)
+
+        protected virtual async Task<string> GetNextChildCodeAsync(Guid? parentId, CancellationToken cancellationToken = default)
         {
-            var lastChild = await GetLastChildOrNullAsync(parentId);
-            if (lastChild == null)
+            if (!parentId.HasValue)
```

Comment:
In original design(like abp `OrganizationUnit`),it allowed to store multiple root nodes ,
all the parentId of  root node is null but code is different, I should keep this feature
Example:
00001.00001
00002.00001
both 00001 and  00002 parentId=null

Reply:
Hi~ @yinchang0626 . I have no intention of changing the original logic of this block of code, I just refactor this for intuitive. Which line of code is not your original design?

Reply:
Hi, thanks for your refactor. the code becomes more beautiful because of you
I have integrated that ,so doesn't matter now
```csharp
protected virtual async Task<string> GetNextChildCodeAsync(Guid? parentId)
  var lastChild = await GetLastChildOrNullAsync(parentId);//this will call sql to check are there others root exist
  if (lastChild == null)
  {	         
                var parentCode = parentId != null ? await GetCodeAsync(parentId.Value) : null;
                return TreeCodeGenerator.AppendCode(parentCode, TreeCodeGenerator.CreateCode(1));	
  }
  return TreeCodeGenerator.CalculateNextCode(lastChild.Code);
}
```
```csharp
protected virtual async Task<string> GetNextChildCodeAsync(Guid? parentId)
 {
  if (!parentId.HasValue)
  {
   return TreeCodeGenerator.Create(1);//always create '00001' if (!parentId.HasValue)
  }
}
``